OpenRA.Game/Graphics/Vertex.cs

```diff
@@ -16,17 +16,34 @@ namespace OpenRA.Graphics
 	[StructLayout(LayoutKind.Sequential)]
 	public struct Vertex
 	{
-		public readonly float X, Y, Z, S, T, U, V, P, C;
+		// 3d position
+		public readonly float X, Y, Z;
+
+		// Primary and secondary texture coordinates or RGBA color
+		public readonly float S, T, U, V;
+
+		// Palette and channel flags
+		public readonly float P, C;
+
+		// Color tint
+		public readonly float R, G, B;
 
 		public Vertex(float3 xyz, float s, float t, float u, float v, float p, float c)
-			: this(xyz.X, xyz.Y, xyz.Z, s, t, u, v, p, c) { }
+			: this(xyz.X, xyz.Y, xyz.Z, s, t, u, v, p, c, new float3(1, 1, 1)) { }
+
+		public Vertex(float3 xyz, float s, float t, float u, float v, float p, float c, float3 tint)
+			: this(xyz.X, xyz.Y, xyz.Z, s, t, u, v, p, c, tint.X, tint.Y, tint.Z) { }
+
+		public Vertex(float x, float y, float z, float s, float t, float u, float v, float p, float c, float3 tint)
+			: this(x, y, z, s, t, u, v, p, c, tint.X, tint.Y, tint.Z) { }
 
-		public Vertex(float x, float y, float z, float s, float t, float u, float v, float p, float c)
+		public Vertex(float x, float y, float z, float s, float t, float u, float v, float p, float c, float r, float g, float b)
 		{
 			X = x; Y = y; Z = z;
 			S = s; T = t;
 			U = u; V = v;
 			P = p; C = c;
+			R = r; G = g; B = b;
```

Comment:
In the next PR (or maybe this one when I rework the palette AllowTint stuff) I expect to add the alpha channel here too - this gives us the plumbing for per-sprite transparency, which will let us remove the palette-swapping translucency hacks from, and therefore add RGBA support to, the building placement previews / ts cloak etc.